patches/minecraft/net/minecraft/client/multiplayer/ClientConfigurationPacketListenerImpl.java.patch

```diff
@@ -1,9 +1,20 @@
 --- a/net/minecraft/client/multiplayer/ClientConfigurationPacketListenerImpl.java
 +++ b/net/minecraft/client/multiplayer/ClientConfigurationPacketListenerImpl.java
-@@ -143,6 +_,7 @@
+@@ -121,7 +_,7 @@
+         );
+         this.connection
+             .setupInboundProtocol(
+-                GameProtocols.CLIENTBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(registryaccess$frozen)),
++                GameProtocols.CLIENTBOUND_TEMPLATE.bind(net.minecraftforge.network.IForgeRegistryFriendlyByteBuf.decorator(registryaccess$frozen, this.connection)),
+                 new ClientPacketListener(
+                     this.minecraft,
+                     this.connection,
+@@ -142,7 +_,8 @@
+                 )
              );
          this.connection.send(ServerboundFinishConfigurationPacket.INSTANCE);
-         this.connection.setupOutboundProtocol(GameProtocols.SERVERBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(registryaccess$frozen)));
+-        this.connection.setupOutboundProtocol(GameProtocols.SERVERBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(registryaccess$frozen)));
++        this.connection.setupOutboundProtocol(GameProtocols.SERVERBOUND_TEMPLATE.bind(net.minecraftforge.network.IForgeRegistryFriendlyByteBuf.decorator(registryaccess$frozen, this.connection)));
```

Comment:
i prefer adding the decorator method to RFBB directly so these patches are smaller, and deprecating the old one so people know to use the new

Reply:
Ok